server.py

```
@@ -14,42 +14,93 @@
     except Exception as e:
         logger.error(f"Error initializing Reddit client: {e}")
         return None
 
 # Initialize Reddit client
-reddit = get_reddit_client()
+_reddit_client = get_reddit_client()
+
+def require_write_access(func):
+    """Decorator to ensure write access is available"""
+    def wrapper(*args, **kwargs):
+        if reddit_is_read_only:
```

Ruff (0.11.9)

Undefined name `reddit_is_read_only`

(F821)